README.md

```diff
@@ -1,39 +1,69 @@
+This project is a FHIR server which is used for multiple actors (EHR and Knowledge Artifact Repository) in the [MedMorph Public Health Usecase](https://build.fhir.org/ig/HL7/fhir-medmorph/usecases.html). It is based on the [HAPI FHIR JPA Server (Version 4.1.0)](https://github.com/hapifhir/hapi-fhir-jpaserver-starter).
 
 Note: Issues related to performance and Subscriptions were observed on HAPI 5.1.0
 
 # Running Locally
 
-The easiest way to run this server is to use docker. First, clone this
-repository. Then, from the repository root run:
+The easiest way to run this server is to use docker.
 
 ```
-docker build -t medmorph_ehr .
+./build-docker-image.sh
+docker-compose up
 ```
 
-This will build the docker image for the reference server. Once the image has
-been built, the server can be run with the following command:
+This will create the `medmorph_fhir` image and spin up the `medmoprh_ehr` (on [http://localhost:8080/fhir](http://localhost:8080/fhir)) and `knowledge_artifact` (on [http://localhost:8090/fhir](http://localhost:8090/fhir)) containers.
+
+Alternatively the image can be built and specific instances run. First, clone this repository. Then, from the repository root run:
 
 ```
-docker run -p 8080:8080 medmorph_ehr
+docker build -t medmorph_fhir .
 ```
 
-Alternatively the image can be built and run using
+This will build the docker image for the reference server. Once the image has
+been built, the server can be run with the following command:
 
 ```
-./build-docker-image.sh
-docker-compose up
+docker run -p 8080:8080 medmorph_fhir -e AUTH_SERVER_ADDRESS=http://moonshot-dev.mitre.org:8090/auth/realms/ehr/protocol/openid-connect/ -e SERVER_ADDRESS=http://localhost:8080/fhir/
 ```
 
 The server will then be browseable at
-[http://localhost:8080/medmorph](http://localhost:8080/medmorph), and the
+[http://localhost:8080](http://localhost:8080), and the
 server's FHIR endpoint will be available at
-[http://localhost:8080/medmorph/fhir](http://localhost:8080/medmorph/fhir)
+[http://localhost:8080/fhir](http://localhost:8080/fhir)
+
+# Configuration
+
+Since this is a single repository and single docker image for multiple actors, each container should have its own properties. Right now the properties to configurate are `AUTH_SERVER_ADDRESS`, `SERVER_ADDRESS`, and `SERVER_TITLE`. These are set through enviornment variables of the same names. Using compose will automatically create all necessary servers.
 
 # Authorization
 
 This server is protected by the SMART Backend Authentication protocol. The admin token is `admin`. The OAuth URLS can be found at the `/metadata` or `/.well-known/smart-configuration` endpoints.
 
-Follow the [Register New Client](https://github.com/mcode/medmorph-ehr/wiki/Register-New-Client) wiki page to register a new client.
+Follow the [Register New Client](https://github.com/mcode/medmorph-fhir-server/wiki/Register-New-Client) wiki page to register a new client.
+
+# Hosted Server
```

Comment:
Can we move this section to a wiki page? I feel like we don't want to list out the details of an internal instance on the public GitHub page. Not that it's a security risk, pathways.mitre.org isn't accessible externally at all, it's just not broadly meaningful. And then if we change things around like ports or URLs we don't have to issue PRs for the readme.

Also, probably worth pointing out that the ports and URLs are already modified in the docker-compose.yml on the server, so a `git pull` might fail if that file is changed in the repo. One option is to git stash, pull, then git stash pop, rebuild, etc.

Reply:
https://github.com/mcode/medmorph-fhir-server/wiki/MITRE-Internal-Server